Refactoring gaussian_filter and gabor_filter (Ref: #1647)

sumitbinnani opened the pull request:
Refactoring ```gaussian_filter``` to ```gaussian```
Refactoring ```gabor_filter``` to ```gabor```
Reference #1647

ahojnnes commented on skimage/filters/tests/test_gaussian.py:
```diff
@@ -37,13 +37,13 @@ def test_multichannel():
     # Iterable sigma
-    gaussian_rgb_a = gaussian_filter(a, sigma=[1, 2], mode='reflect',
+    gaussian_rgb_a = gaussian(a, sigma=[1, 2], mode='reflect',
                                      multichannel=True)
```
Make sure to adapt the indentation everywhere, please.

sumitbinnani replied:
@ahojnnes Sorry, I am unable to your statement :-(

Can you please elaborate a bit?

emmanuelle replied:
It's a PEP8 style point about indentation, see https://www.python.org/dev/peps/pep-0008/#indentation
Here the second line or arguments should be aligned with the first one, so ``multichannel=True`` should be just under `a`.

sumitbinnani commented:
@emmanuelle : The built is failing for Python 3.2. The issue existed even when no changes were incorporated. Kindly guide.

sumitbinnani commented:
@ahojnnes @stefanv 
Should I close this PR and open a new one with properly grouped commit?

blink1073 commented:
@sumitbinnani, you can regroup these commits by rebasing against master:

```
git fetch upstream master
git rebase -i upstream/master
```

Then, you can change `pick` to `s` or `squash` to combine commits.

blink1073 commented:
When you've finished, you `git push --force origin RenamingFilter`

sumitbinnani commented:
@blink1073, Thanks for the help :-)

blink1073 commented:
No problem @sumitbinnani.  The 3.2 problem is a known issue, and we are trying to figure out what to do about it in #1740.

sumitbinnani commented:
Added alias and deprecation warning for ```skimage.filters.gabor_filter``` and ```skimage.filters.gaussian_filter```. Also, updated ```TODO.txt```. 

Kindly let me know if any changes are required.

Thanks, Sumit

sumitbinnani commented:
Updated edited files to comply with PEP8 Standards.

sumitbinnani commented:
@blink1073 @stefanv 
The code coverage drop after making changes to comply with PEP8 Standards (may be an extra line got added due to it). Build 3.2 passed but 3.5 failed. Kindly guide.

blink1073 commented:
The Travis error looks to be unrelated to your PR.  I restarted it.  If it fails, I'll raise it as a separate issue.

blink1073 commented:
Travis is happy, looks good to me, but I'll let @stefanv press the green button.

sumitbinnani commented:
Hi @blink1073 @stefanv 

Does the PR need more changes? Kindly let me know if additional changes are required :-)

blink1073 commented:
@stefanv, any problem with merging this?

stefanv commented:
Please, go ahead.

blink1073 commented:
Done, thanks @sumitbinnani!

sumitbinnani commented:
Happy to contribute :-)
